pkg/pillar/zfs/zfs.go

```diff
@@ -91,41 +90,24 @@ func DestroyDataset(log *base.LogObject, dataset string) (string, error) {
 	return string(stdoutStderr), err
 }
 
-//GetDatasetOptions get dataset options from zfs
-//will return error if not exists
-func GetDatasetOptions(log *base.LogObject, dataset string) (map[string]string, error) {
-	args := []string{"get", "-Hp", "-o", "property,value", "all", dataset}
-	stdoutStderr, err := base.Exec(log, types.ZFSBinary, args...).CombinedOutput()
+// DatasetExist return true if dataset exists or false when it does not exist
+func DatasetExist(datasetPath string) bool {
```

Comment:
DatasetExist() will return false when the Dataset doesn't exist or in case of errors, how we could differentiate false due to non-existent Dataset from other errors?